src/core/containers/runtime/uv_provider.py

```diff
@@ -0,0 +1,183 @@
+"""Providers for launching Hugging Face Spaces via ``uv run``."""
+
+from __future__ import annotations
+
+import os
+import socket
+import subprocess
+import time
+from dataclasses import dataclass, field
+from typing import Dict, Optional
+
+import requests
+
+from .providers import ContainerProvider
+
+
+def _poll_health(health_url: str, timeout_s: float) -> None:
+    """Poll a health endpoint until it returns HTTP 200 or times out."""
+
+    deadline = time.time() + timeout_s
+    while time.time() < deadline:
+        try:
+            response = requests.get(health_url, timeout=2.0)
+            if response.status_code == 200:
+                return
+        except requests.RequestException:
+            pass
+
+        time.sleep(0.5)
+
+    raise TimeoutError(
+        f"Server did not become ready within {timeout_s:.1f} seconds"
+    )
+
+
+def _create_uv_command(
+    repo_id: str,
+    host: str,
+    port: int,
+    reload: bool,
+    project_url: Optional[str] = None,
+) -> list[str]:
+    command = [
+        "uv",
+        "run",
+        "--project",
+        project_url or f"git+https://huggingface.co/spaces/{repo_id}",
+        "--",
+        "server",
+        "--host",
+        host,
+        "--port",
+        str(port),
+    ]
+    if reload:
+        command.append("--reload")
+    return command
+
+
+@dataclass
+class UVProvider(ContainerProvider):
```

Comment:
```suggestion
class UVRuntime(RuntimeProvider):
```

 (naming suggestion to be aligned with the "runtime/" folder)